src/main/scala/api/sampling/SamplingRegistration.scala

```diff
@@ -80,7 +78,7 @@ class SamplingRegistration(model: StatisticalMeshModel, sample: TriangleMesh3D,
         val bestTheta = bestSamplelogger.currentBestSample().get
```

Comment:
Maybe you should do proper error handling here (get will just throw an exception if things don't work as expected)